### ISSUE TITLE
New Migration to IATI Publisher Request

### ISSUE THREAD

**Issue by DaveEade49** [younginnovations/iatipublisher#1277]
The following organisation has requested to migrated to IATI Publisher:

REPORTING ORGANISATION
IMPACT Initiatives 
ORGANISATION IDENTIFIER
XM-OCHA-HPC9829

Request came via Zendesk: https://iati.zendesk.com/agent/tickets/40134

**Comment by Sanilblank** [comment]
@DaveEade49 I looked into the organisation which needs to be migrated to IATI Publisher and found that 'IMPACT initiatives' has already created an account on IATI Publisher. We could migrate the organisation using the 'existing organisation migration' script but the script does not account for the organisation information i.e. the organisation information from AidStream will not be migrated.
This may cause issues as the said organisation has input some organisation data on AidStream and if we use the 'existing organisation migration' script to migrate the organisation the information relating to activity, users, and such will be migrated but the organisation information will be skipped.
Looking into the account they have created on IATI Publisher, it seems they have not input any proper information yet so if the organisation information present on AidStream is important to be migrated, we could just delete the organisation from IATI Publisher and then migrate the organisation as a new organisation. If this is not feasible, we will either have to update the script or just migrate the organisation skipping the organisation information, and then the user will have to input the organisation information themselves.

cc. @praweshsth

**Comment by DaveEade49** [comment]
Hi, yes that's fine - migrate as it is ( I will get the organisation data updated in publisher ).
Thanks
Dave
________________________________

**Comment by DaveEade49** [comment]
@Sanilblank - Hi, can you tell me when Migration will take place - I've had a follow up from the organisation. Thanks Dave

**Comment by DaveEade49** [comment]
@Sanilblank @praweshsth - Just looked at their account in publisher - they have 3 activities already, and 5 in Aidstream - I need to speak to them first before we do anything.

**Comment by Sanilblank** [comment]
@DaveEade49 Ok. We will perform the migration once you confirm what we need to do.

**Comment by DaveEade49** [comment]
Hi, can we migrate this using the 'existing organisation migration' script.

Will this mean, that the activities they have already created in Publisher will remain?

Is it an option to download the XML from Aidstream for the activities and upload into publisher?

Thanks
David

**Comment by DaveEade49** [comment]
@Sanilblank - see comments above

**Comment by Sanilblank** [comment]
When migrating the organisation using the existing migration script, the activities present on AidStream which have the same identifier as the activities present on IATI Publisher will not be migrated. Any other activities will be migrated. Similar case for users as well, users having existing emails will not be migrated but the others will be migrated.

Yes, you can also download the XML file from AidStream and then upload them to IATI Publisher (The combined XML file will have to be downloaded from the registry since AidStream does not have the feature to download the combined XML file). When uploading the XML file downloaded in this way, the existing activities having the same identifiers as of that present in the file, will have to be either updated or skipped.
@DaveEade49

**Comment by DaveEade49** [comment]
Thanks - can you confirm you are progressing with the migration? @Sanilblank

**Comment by Sanilblank** [comment]
@DaveEade49 Did you mean to say to start the migration process using the existing migration script?

**Comment by DaveEade49** [comment]
yes please @Sanilblank

**Comment by Sanilblank** [comment]
@DaveEade49 We performed the migration on the development server for testing and found that two of their activities had budget validation errors.

Activity id :XM-OCHA-HPC9829-35iAHK
Budget: Budget with type revised must have period start and end same to that of one of the budgets having same type original for budgets elements at position 4 
Activity id : XM-OCHA-HPC9829-32iAIE
Budget with type revised must have period start and end same to that of one of the budgets having same type original for budgets elements at position 1, 2, 3

Can we move ahead with the migration or do we need to update the data?

**Comment by DaveEade49** [comment]
yes, migrate but remove the dates that are invalid.

**Comment by Sanilblank** [comment]
@DaveEade49 Removing the dates on AidStream is not possible as the dates are a required field when adding the budget. We can remove the entire budget revised part that is causing the issue on AidStream and then migrate the data but since the activity is a published one, this will not affect the data that had been previously published and will update the data only on AidStream.

Another option is we can just migrate the data as it is to IATI Publisher and then remove the budget revised part that is causing the issues (removing the dates for only the revised type is not possible here as well) but this will be the same as the AidStream part (actual published data will remain the same and the updated at date will change). We can then publish the activities but the data and dates will change which may be an issue.

So, we are confused about how to move forward with this.

**Comment by DaveEade49** [comment]
ah, ok yes I remember now - can you just add them as "original" rather than "revised" @Sanilblank

**Comment by Sanilblank** [comment]
@DaveEade49 Yeah we can do that. But changing the data from the form will update the activity updated at date. Also, since we won't be publishing the activity, the activity present on the IATI registry will not reflect the change and will contain the old data. Is this ok?

**Comment by Sanilblank** [comment]
@DaveEade49 We have recently migrated the AidStream server to a different location so we need to update the credentials in the IATI Publisher. Also, changing a certain part of the code for other features has resulted in a minor issue during the migration process, so we will be looking into these and then migrating this and the other organisation mentioned in issue no #1284 at same time if possible.

**Comment by DaveEade49** [comment]
@Sanilblank - ok thats great thanks

**Comment by sriza** [comment]
@DaveEade49 
We have migrated `IMPACT Initiatives` to the production site.
Please have a look.

Regarding the migration of Aidstream server and database, we have updated the new credentials of the server in PR #1288 and have deployed it to production.

Thank you
